app/api/common.py

```diff
@@ -26,25 +26,29 @@ def states_daily_query(state=None, preview=False):
         filter_list.append(Batch.isPublished == True)
 
     latest_state_daily_batches = db.session.query(
-        CoreData.state, CoreData.date, func.max(CoreData.batchId).label('maxBid')
-        ).join(Batch
-        ).filter(*filter_list
-        ).group_by(CoreData.state, CoreData.date
-        ).subquery('latest_state_daily_batches')
+        CoreData.state, CoreData.date, func.max(CoreData.batchId).label('maxBid'),
+        func.row_number().over(
```

Comment:
Could you add a comment about what the `func.row_number()...partition_by` is doing? It's an increasingly scary sql query :p

Reply:
`OVER`/`PARTITION BY` is like `group by` in the sense that it groups the result, but there's no aggregation, so over each partition there's an assigned `row_number` function, but it's not being aggregated into anything. In the external query, the row_number is used to filter older rows.
So the result is the most recent `n` rows for each state.


We don't expect to have weird dating issues, but this will handle weird dating issues in a reasonable way.

Reply:
thanks! that makes sense - could you put some version of this explanation into a code comment above this query?